public/app/features/alerting/unified/api/alertmanager.ts

```diff
@@ -37,3 +44,75 @@ export async function updateAlertmanagerConfig(
     config
   );
 }
+
+export async function fetchSilences(alertmanagerSourceName: string): Promise<Silence[]> {
+  const result = await getBackendSrv()
+    .fetch<Silence[]>({
+      url: `/api/alertmanager/${getDatasourceAPIId(alertmanagerSourceName)}/api/v2/silences`,
+      showErrorAlert: false,
+      showSuccessAlert: false,
+    })
+    .toPromise();
+  return result.data;
+}
+
+//returns silence id
+export async function createOrUpdateSilence(
+  alertmanagerSourceName: string,
+  payload: SilenceCreatePayload
+): Promise<string> {
+  const result = await getBackendSrv().post(
+    `/api/alertmanager/${getDatasourceAPIId(alertmanagerSourceName)}/api/v2/silences`,
+    payload
+  );
+  return result.silenceID;
+}
+
+export async function expireSilence(alertmanagerSourceName: string, silenceID: string): Promise<void> {
+  await getBackendSrv().delete(
+    `/api/alertmanager/${getDatasourceAPIId(alertmanagerSourceName)}/api/v2/silences/${encodeURIComponent(silenceID)}`
+  );
+}
+
+export async function fetchAlerts(
+  alertmanagerSourceName: string,
+  matchers?: SilenceMatcher[]
+): Promise<AlertmanagerAlert[]> {
+  const filters =
+    matchers
+      ?.map(
+        (matcher) =>
+          `filter=${encodeURIComponent(
+            `${escapeQuotes(matcher.name)}=${matcher.isRegex ? '~' : ''}"${escapeQuotes(matcher.value)}"`
+          )}`
+      )
+      .join('&') || '';
+
+  const result = await getBackendSrv()
+    .fetch<AlertmanagerAlert[]>({
+      url:
+        `/api/alertmanager/${getDatasourceAPIId(alertmanagerSourceName)}/api/v2/alerts` +
+        (filters ? '?' + filters : ''),
+      showErrorAlert: false,
+      showSuccessAlert: false,
+    })
+    .toPromise();
+
+  return result.data;
+}
+
+export async function fetchAlertGroups(alertmanagerSourceName: string): Promise<AlertmanagerGroup[]> {
+  const result = await getBackendSrv()
+    .fetch<AlertmanagerGroup[]>({
+      url: `/api/alertmanager/${getDatasourceAPIId(alertmanagerSourceName)}/api/v2/alerts/groups`,
+      showErrorAlert: false,
+      showSuccessAlert: false,
+    })
+    .toPromise();
+
+  return result.data;
+}
+
+function escapeQuotes(value: string): string {
+  return value.replace(/"/g, '\\"');
```

Comment:
nit: looks like we are missing a `/` at the end here.

Reply:
this is entirely intentional, it replaces `"` with `\"`

Reply:
ah, mybad 🙈 I was a bit quick looking through this.